pom.xml

```diff
@@ -217,19 +217,29 @@
           <groupId>org.apache.httpcomponents</groupId>
           <artifactId>httpmime</artifactId>
         </exclusion>
+        <exclusion>
+          <groupId>com.atlassian.httpclient</groupId>
+          <artifactId>atlassian-httpclient-plugin</artifactId>
+        </exclusion>
       </exclusions>
     </dependency>
     <dependency>
       <groupId>com.atlassian.fugue</groupId>
       <artifactId>fugue</artifactId>
       <version>${fugue.version}</version>
-      <scope>runtime</scope>
     </dependency>
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>apache-httpcomponents-client-4-api</artifactId>
       <version>4.5.3-2.0</version>
     </dependency>
+
+    <dependency>
+      <groupId>org.apache.httpcomponents</groupId>
```

Comment:
Shoudl this perhaps be https://wiki.jenkins.io/display/JENKINS/Async+Http+Client+Plugin cc @vivek

Reply:
This one is based on 
```
      <groupId>com.ning</groupId>
      <artifactId>async-http-client</artifactId>
      <version>1.9.40</version>
```

Reply:
ah right - maybe I was thinking of https://wiki.jenkins.io/display/JENKINS/Apache+HttpComponents+Client+4.x+API+Plugin - does that include async?

Reply:
@michaelneale https://wiki.jenkins.io/display/JENKINS/Apache+HttpComponents+Client+4.x+API+Plugin does not include async part right now.

Reply:
@olamy 
1. The 4.1.3 version of org.apache.httpcomponents:httpasyncclient does not have any package level dependency on other part of httpcomponents, but since Apache httpcomponents maintainers have freedom to change the design, can you confirm that this will not happen in future? Or another java.lang.IllegalAccessError might occur when httpasyncclient upgraded.
2. Rewrite HTTP staff to use https://wiki.jenkins.io/display/JENKINS/Async+Http+Client+Plugin would be great :)

Reply:
@olamy No problem now since https://github.com/jenkinsci/apache-httpcomponents-client-4-api-plugin/pull/6 has been merged. 👍